### PR TITLE
fix: columns and filters for sales analytics report

### PR DESCRIPTION
**Pull request by harshmule-epi**
* Asana Task Link:

https://app.asana.com/0/1200471303892628/1200402704793206/f

* Updated Report
Screenshot 2021-06-29 at 6 59 15 PM

### CODE REVIEW

File: erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py [hunk 1]
```diff
@@ -30,13 +30,7 @@ def run(self):
...
-				"label": _(self.filters.tree_type + " ID"),
-				"options": self.filters.tree_type if self.filters.tree_type != "Order Type" else "",
-				"fieldname": "entity",
-				"fieldtype": "Link" if self.filters.tree_type != "Order Type" else "Data",
-				"width": 140 if self.filters.tree_type != "Order Type" else 200
-			}]
+		self.columns = []
 
 		if self.filters.tree_type in ["Customer", "Supplier", "Item"]:
 			self.columns.append({
```
Comment (hrwX) [inline erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py:36]: - If this is Sales Report by Customer, there should a link field for customer in here

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py (hunk 1 above)
Comment (harshmule-epi) [reply erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py:36]: Where, it is already there I think!

Comment (harshmule-epi) [reply erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py:36]: It's actually consist of Item and Customer both that's why it's not specific to link to customer!
and the name was mentioned as Sales Report by Customer, that's why i changed it to Sales Report By Customer

File: erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py [hunk 2]
```diff
@@ -292,11 +285,11 @@ def get_periodic_data(self):
 
 	def get_period(self, posting_date):
 		if self.filters.range == 'Weekly':
-			period = "Week " + str(posting_date.isocalendar()[1]) + " " + str(posting_date.year)
+			period = "Week " + str(posting_date.isocalendar()[1]) + ", " + str(posting_date.year)
 		elif self.filters.range == 'Monthly':
 			period = str(self.months[posting_date.month - 1]) + " " + str(posting_date.year)
 		elif self.filters.range == 'Quarterly':
-			period = "Quarter " + str(((posting_date.month - 1) // 3) + 1) + " " + str(posting_date.year)
+			period = "Q" + str(((posting_date.month - 1) // 3) + 1) + " " + str(posting_date.year)
```
Comment (rvpasari) [inline erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py]: Add a comma, should look like Q1, 2020

**Review by rvpasari — CHANGES_REQUESTED** [review]

**Comment by rvpasari** [comment]
@harshmule-git can you post a screen shot of how the new report looks like right now?

@sahil28297 we can merge changes for this report, there is no need to specify domain over here

File: erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py (hunk 2 above)
Comment (harshmule-epi) [reply erpnext/selling/report/sales_report_by_customer/sales_report_by_customer.py]: okay

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**